Automattic/wp-calypso
# Updates the promote post widget header

Related to SELFSERVE-586

## Proposed Changes

We are updating the style of the Create campaign to the one designed for the Iteration 2 of the Dashboard. We are maintaining compatibility with version 1 here, so you should be able to see both version by disabling the corresponding feature flag.

For this iteration, I am reusing the BlankCanvas.Header component. I had to add the possibility to get a `className` and also added typescript support on those components.

**Previous:**
![Screenshot 2023-07-07 at 11 17 41](https://github.com/Automattic/wp-calypso/assets/1258162/2e2b2dca-8b7b-455e-8cb5-ed6274f4b86e)

**New:**
![Screenshot 2023-07-07 at 11 16 46](https://github.com/Automattic/wp-calypso/assets/1258162/b324c65e-a66b-461d-a5cc-f92434b26a34)

## Testing Instructions

* Open the live preview and navigate to Tools->Advertising
* Click on the Promote button (in the header or of a post)
* Verify that the new version of the header is shown
* Cancel the widget and disable the flag `promote-post/widget-i2` in the URL
* Click on the Promote button again and verify that the old design is shown correctly

## Pre-merge Checklist
